x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/support/mapper/CompositeRoleMapperTests.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.security.authc.support.mapper;
+
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.action.support.PlainActionFuture;
+import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.xpack.core.security.authc.support.CachingRealm;
+import org.elasticsearch.xpack.core.security.authc.support.UserRoleMapper;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+
+import static org.elasticsearch.test.ActionListenerUtils.anyActionListener;
+import static org.hamcrest.Matchers.equalTo;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+public class CompositeRoleMapperTests extends ESTestCase {
+
+    public void testClearRealmCachePropagates() {
+        UserRoleMapper userRoleMapper1 = mock(UserRoleMapper.class);
+        UserRoleMapper userRoleMapper2 = mock(UserRoleMapper.class);
+        CompositeRoleMapper compositeRoleMapper = new CompositeRoleMapper(userRoleMapper1, userRoleMapper2);
+        CachingRealm realm = mock(CachingRealm.class);
+        compositeRoleMapper.clearRealmCacheOnChange(realm);
+        verify(userRoleMapper1, times(1)).clearRealmCacheOnChange(eq(realm));
+        verify(userRoleMapper1, times(1)).clearRealmCacheOnChange(eq(realm));
```

Comment:
I think you meant:
```suggestion
        verify(userRoleMapper1, times(1)).clearRealmCacheOnChange(eq(realm));
        verify(userRoleMapper2, times(1)).clearRealmCacheOnChange(eq(realm));
```

?

Reply:
Good catch! Thanks!